### ISSUE TITLE
hi ,tThank you for sharing. Now I have some problems and want to ask you. The code runs to this step, but I don’t know where the problem is. Can you explain it to me, thank youhanksyou

### ISSUE THREAD

**Issue by liuzhiwei1997** [jackyko1991/vnet-tensorflow#18]
![image](https://user-images.githubusercontent.com/76031710/122701383-b7b98380-d27f-11eb-95dd-384beb6535aa.png)
You can see that my global step is always 0, and the loss is also 0

**Comment by jackyko1991** [comment]
epoch loss should not be zero, please check the tensorflow version you are using

**Comment by liuzhiwei1997** [comment]
Thank you for your reply, I used tensorflow-gpu=2.4.0 and 2.1.0，




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jackyko1991/vnet-tensorflow"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月21日(星期一) 下午5:17
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jackyko1991/vnet-tensorflow] hi ,tThank you for sharing. Now I have some problems and want to ask you. The code runs to this step, but I don’t know where the problem is. Can you explain it to me, thank youhanksyou (#18)





 
epoch loss should not be zero, please check the tensorflow version you are using

**Comment by jackyko1991** [comment]
Please use tensorflow 1.15

**Comment by liuzhiwei1997** [comment]
Ok thank you, let me try------------------&nbsp;原始邮件&nbsp;------------------
***@***.***&gt;
发送时间:&nbsp;2021年6月21日(星期一) 下午5:34
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jackyko1991/vnet-tensorflow] hi ,tThank you for sharing. Now I have some problems and want to ask you. The code runs to this step, but I don’t know where the problem is. Can you explain it to me, thank youhanksyou (#18)

**Comment by liuzhiwei1997** [comment]
Hello,sir， I just tried the 1.15 version of tensorflow, but nothing has changed. The global step and epoch loss are still 0，Do you think there might be something wrong?Here are two pictures you can see



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jackyko1991/vnet-tensorflow"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月21日(星期一) 下午5:34
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jackyko1991/vnet-tensorflow] hi ,tThank you for sharing. Now I have some problems and want to ask you. The code runs to this step, but I don’t know where the problem is. Can you explain it to me, thank youhanksyou (#18)





 
Please use tensorflow-gpu 1.15

**Comment by liuzhiwei1997** [comment]
I tried your newly uploaded code, and a new problem appeared, as shown in the figure

Do you know what's going on，thankyou


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jackyko1991/vnet-tensorflow"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月21日(星期一) 下午5:34
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jackyko1991/vnet-tensorflow] hi ,tThank you for sharing. Now I have some problems and want to ask you. The code runs to this step, but I don’t know where the problem is. Can you explain it to me, thank youhanksyou (#18)





 
Please use tensorflow-gpu 1.15

**Comment by jackyko1991** [comment]
Set `"ImageLog": true`, in `config.json`, start tensorboard to check if the images are loaded properly

**Comment by liuzhiwei1997** [comment]
Hello, I still haven’t solved the loss problem. I checked the picture on tensorboard. I don’t know where the problem is. I hope you can help me.Is it because loss is always 0, so there are only graphs in tensorboard, no other data charts





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jackyko1991/vnet-tensorflow"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月22日(星期二) 上午10:27
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jackyko1991/vnet-tensorflow] hi ,tThank you for sharing. Now I have some problems and want to ask you. The code runs to this step, but I don’t know where the problem is. Can you explain it to me, thank youhanksyou (#18)





 
Set "ImageLog": true, in config.json, start tensorboard to check if the images are loaded properly

**Comment by jackyko1991** [comment]
Please check if your data is loaded properly

**Comment by liuzhiwei1997** [comment]
Is it convenient to ask about the specific form of your data, I am using 512x512x22 MRI images now


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jackyko1991/vnet-tensorflow"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月23日(星期三) 上午10:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jackyko1991/vnet-tensorflow] hi ,tThank you for sharing. Now I have some problems and want to ask you. The code runs to this step, but I don’t know where the problem is. Can you explain it to me, thank youhanksyou (#18)





 
Please check if your data is loaded properly

**Comment by liuzhiwei1997** [comment]
I tried your updated program, this error occurred, may I ask what is wrong with the data set







------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jackyko1991/vnet-tensorflow"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年6月23日(星期三) 上午10:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [jackyko1991/vnet-tensorflow] hi ,tThank you for sharing. Now I have some problems and want to ask you. The code runs to this step, but I don’t know where the problem is. Can you explain it to me, thank youhanksyou (#18)





 
Please check if your data is loaded properly

**Comment by liuzhiwei1997** [comment]
1:The first is why some slices are discarded when reading slices, as shown in the program





2:The second question is loss. When using mixed loss, why the parameter alpha only constrains one of the losses, and when alpha=1, it means whether this parameter is redundant



Hope to hear from you, thank you

**Comment by jackyko1991** [comment]
> 1:The first is why some slices are discarded when reading slices, as shown in the program 2:The second question is loss. When using mixed loss, why the parameter alpha only constrains one of the losses, and when alpha=1, it means whether this parameter is redundant Hope to hear from you, thank you

1. If you are specifying 2D segmentation cases, the occurence rate of empty label slice may affect training results. I have set a probability to discard part of the slices to gaurantee the training slices contains certain size of segmentation labels. E.g. A CT scan may only contain few pixels across few slices out of thousands of scanning images, the training will be easily dominated by zero labels as you hardly let the model access to pixels with desired regions.

2. The regularization only need to constrain on cross entropy loss as DICE loss is bounded in [0,1]. Alpha is to restrict the percentage of contribution during loss optimization. This factor is subjected to your dataset, setting alpha = 1 is an arbitary choice. To test regularization parameter, you may conduct trial to obtain testing score across different alpha values, e.g.:
[![alpha](https://scikit-survival.readthedocs.io/en/stable/_images/user_guide_survival-svm_41_0.png)](https://scikit-survival.readthedocs.io/en/stable/user_guide/survival-svm.html)
This plot indicates best model performance with alpha  = 0.01562, whereas a robust alogrithm/ dataset would have less disperse test scores:
[![alpha](https://scikit-survival.readthedocs.io/en/stable/_images/user_guide_survival-svm_19_0.png)](https://scikit-survival.readthedocs.io/en/stable/user_guide/survival-svm.html)

**Comment by liuzhiwei1997** [comment]
Thank you for your guidance, so do I need to adjust the alpha by myself now to get the best results? I will try my best。In addition, I have another question about the weight in weighted_sorensen loss. You set [0.01,0.1]. I don’t understand how weight affects the calculation, so I don’t know how to adjust it. I hope you can help me answer it, thank you




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Ka ***@***.***&gt;; 
发送时间: 2021年9月13日(星期一) 上午10:36
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [jackyko1991/vnet-tensorflow] hi ,tThank you for sharing. Now I have some problems and want to ask you. The code runs to this step, but I don’t know where the problem is. Can you explain it to me, thank youhanksyou (#18)





  
1:The first is why some slices are discarded when reading slices, as shown in the program 2:The second question is loss. When using mixed loss, why the parameter alpha only constrains one of the losses, and when alpha=1, it means whether this parameter is redundant Hope to hear from you, thank you
   
 
If you are specifying 2D segmentation cases, the occurence rate of empty label slice may affect training results. I have set a probability to discard part of the slices to gaurantee the training slices contains certain size of segmentation labels. E.g. A CT scan may only contain few pixels across few slices out of thousands of scanning images, the training will be easily dominated by zero labels as you hardly let the model access to pixels with desired regions.
 
 
 
The regularization only need to constrain on cross entropy loss as DICE loss is bounded in [0,1]. Alpha is to restrict the percentage of contribution during loss optimization. This factor is subjected to your dataset, setting alpha = 1 is an arbitary choice. To test regularization parameter, you may conduct trial to obtain testing score across different alpha values, e.g.:
 
 This plot indicates best model performance with alpha  = 0.01562, whereas a robust alogrithm/ dataset would have less disperse test scores:
 
 
  
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by jackyko1991** [comment]
@liuzhiwei1997 You need to adjust alpha and weight to get optimal results. You may observe DICE and cross entropy loss in tensorboard to check if they are in samiliar order of magnitude. For class weight in weighted sorensen loss, it is advise to count 0 labeled segmentation pixels (background) to actual segmentation labels. Usually 0 to 1 label ratio can up to thousands to one.

For quick hyperparameter tuning you may check [Guild AI](https://guild.ai/)
